factorio_rcon/_factorio_rcon.py

```diff
@@ -402,6 +403,19 @@ def send_commands(self, commands: Dict[T, str]) -> Dict[T, Optional[str]]:
             else:
                 results[id_map[response.id]] = response.body.rstrip()
         return results
+    
+    def __enter__(self) -> Self:
```

Comment:
Might make more sense to use `handle_socket_errors()` here as it will raise if not connected